4.cpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <vector>
+#include <cstdlib>
+
+double dRand(double min, double max)
+{
+	return min + (rand() % static_cast<unsigned int>((max - min) * 1000)) / 1000.;
+}
+void fillRandom(double * array, int size)
+{
+	double min = -1.0;
+	double max = 1.0; 
+	for (int i = 0; i < size; i++)
+	{
+		double input = dRand(min, max);
+		array[i] = input;
+	}
+}
+
+void sortByGrow(double * array, int size)
+{
+	for (int j = 0; j < size; j++)
+	{
+		int posit = j;
+		for (int k = j+1; k < size; k++)
+		{
+			if(array[k] < array[posit])
+			{
+				posit = k;
+				array[posit] = array[k];
+			}
+		}
+		std::swap (array[posit], array[j]);
+	}	
+}
+
+void printVector(double * array, int size)
+{
+	for (int i = 0; i < size; ++i)
+	{
+		std::cout << array[i] << "\n";
+	}
+}
+
+int main()
+{	
+	std::vector<double> v1(5);
+	fillRandom(v1.data(), v1.size());	
+	std::cout << "Вектор 1 заполнен числами: \n";
+	printVector(v1.data(), v1.size());
+	sortByGrow(v1.data(), v1.size());	
```

Comment:
narrowing conversion

Reply:
corrected 
![4cppsrc](https://user-images.githubusercontent.com/25636404/30023400-d94996cc-9177-11e7-9052-bf6811fddec1.png)

Reply:
@IgorGorbunov20 No, you did not.
https://msdn.microsoft.com/ru-ru/library/hh279667.aspx

8.5.4/6 of n3225
```
A narrowing conversion is an implicit conversion

from a floating-point type to an integer type, or
from long double to double or float, or from double to float, except where the source is a constant expression and the actual value after conversion is within the range of values that can be represented (even if it cannot be represented exactly), or
from an integer type or unscoped enumeration type to a ﬂoating-point type, except where the source is a constant expression and the actual value after conversion will fit into the target type and will produce the original value when converted back to the original type, or
from an integer type or unscoped enumeration type to an integer type that cannot represent all the values of the original type, except where the source is a constant expression and the actual value after conversion will fit into the target type and will produce the original value when converted back to the original type.
```